game.py

```diff
@@ -64,19 +66,29 @@ def put_down_card_on_table(self, player, card):
 
     def try_playing_matching_number_or_color_card(self, player):
         for card in player.cards:
```

Comment:
instead of using a for loop O(n) every time to go through the cards, you can also try using a different data structure to keep them so that searching for a card by type/color each turn is more efficient O(1).